### PR TITLE
Planner - adding null coalesce

### PR DESCRIPTION
**Pull request by hmerrettICHK**
**Description**
Adding null coalesce across all process files in module

**Motivation and Context**

**How Has This Been Tested?**

**Screenshots**

### CODE REVIEW

File: modules/Planner/planner_addProcess.php [hunk 1]
```diff
@@ -49,22 +49,22 @@
...
             }
-            $gibbonCourseClassID = $_POST['gibbonCourseClassID'];
-            $date = dateConvert($guid, $_POST['date']);
-            $timeStart = $_POST['timeStart'];
-            $timeEnd = $_POST['timeEnd'];
+            $gibbonCourseClassID = $_POST['gibbonCourseClassID'] ?? '';
+            $date = dateConvert($guid, $_POST['date'] ?? '');
+            $timeStart = $_POST['timeStart'] ?? '';
+            $timeEnd = $_POST['timeEnd'] ?? '';
             $gibbonUnitID = !empty($_POST['gibbonUnitID']) ? $_POST['gibbonUnitID'] : null;
```
Comment (SKuipers) [inline modules/Planner/planner_addProcess.php:56]: 👀

File: modules/Planner/planner_deleteProcess.php [hunk 2]
```diff
@@ -20,13 +20,13 @@
...
 
 $gibbonPlannerEntryID = $_GET['gibbonPlannerEntryID'];
-$viewBy = isset($_POST['viewBy'])? $_POST['viewBy'] : '';
-$subView = isset($_POST['subView'])? $_POST['subView'] : '';
+$viewBy = $_POST['viewBy'] ?? '';
+$subView = $_POST['subView'] ?? '';
 if ($viewBy != 'date' and $viewBy != 'class') {
     $viewBy = 'date';
 }
 $gibbonCourseClassID = isset($_POST['gibbonCourseClassID'])? $_POST['gibbonCourseClassID'] : '';
```
Comment (SKuipers) [inline modules/Planner/planner_deleteProcess.php:28]: 👀

File: modules/Planner/planner_deleteProcess.php [hunk 3]
```diff
@@ -20,13 +20,13 @@
 include '../../gibbon.php';
 
 $gibbonPlannerEntryID = $_GET['gibbonPlannerEntryID'];
```
Comment (SKuipers) [inline modules/Planner/planner_deleteProcess.php:22]: 👀

File: modules/Planner/planner_duplicateProcess.php [hunk 4]
```diff
@@ -20,15 +20,15 @@
 include '../../gibbon.php';
 
 $gibbonPlannerEntryID = $_GET['gibbonPlannerEntryID'];
```
Comment (SKuipers) [inline modules/Planner/planner_duplicateProcess.php:22]: 👀

File: modules/Planner/planner_editProcess.php [hunk 5]
```diff
@@ -25,13 +25,13 @@
 
 
 $gibbonPlannerEntryID = $_GET['gibbonPlannerEntryID'];
```
Comment (SKuipers) [inline modules/Planner/planner_editProcess.php:27]: 👀

File: modules/Planner/planner_view_full_submitProcess.php [hunk 6]
```diff
@@ -61,7 +61,7 @@
 
             if ($result->rowCount() != 1) {
                 $URL .= '&return=error2';
-                header("Location: {$URL}");
+                header("Location: {$URL}"); ?? ''
```
Comment (SKuipers) [inline modules/Planner/planner_view_full_submitProcess.php:64]: Hmm. This is not a $_POST variable and will cause a fatal error.

File: modules/Planner/resources_add_ajaxProcess.php [hunk 7]
```diff
@@ -34,19 +34,19 @@
         exit();
     } else {
         //Proceed!
-        $id = $_POST['id'];
-        $type = $_POST[$id.'type'];
+        $id = $_POST['id'] ?? '';
+        $type = $_POST[$id.'type'] ?? '';
         if ($type == 'File') {
             $content = $_FILES[$id.'file'];
```
Comment (SKuipers) [inline modules/Planner/resources_add_ajaxProcess.php:40]: 👀

File: modules/Planner/resources_add_ajaxProcess.php [hunk 8]
```diff
@@ -34,19 +34,19 @@
...
         } elseif ($type == 'Link') {
             $content = $_POST[$id.'link'];
```
Comment (SKuipers) [inline modules/Planner/resources_add_ajaxProcess.php:42]: 👀

File: modules/Planner/resources_add_ajaxProcess.php [hunk 9]
```diff
@@ -34,19 +34,19 @@
...
         }
-        $name = $_POST[$id.'name'];
-        $category = $_POST[$id.'category'];
-        $purpose = $_POST[$id.'purpose'];
+        $name = $_POST[$id.'name'] ?? '';
+        $category = $_POST[$id.'category'] ?? '';
+        $purpose = $_POST[$id.'purpose'] ?? '';
         $tags = strtolower($_POST[$id.'tags']);
```
Comment (SKuipers) [inline modules/Planner/resources_add_ajaxProcess.php:47]: 👀

File: modules/Planner/resources_add_ajaxProcess.php [hunk 10]
```diff
@@ -34,19 +34,19 @@
...
         $gibbonYearGroupIDList = (!empty($_POST[$id.'gibbonYearGroupID']))? implode(',', $_POST[$id.'gibbonYearGroupID']) : '';
```
Comment (SKuipers) [inline modules/Planner/resources_add_ajaxProcess.php:48]: 👀

File: modules/Planner/resources_manage_addProcess.php [hunk 11]
```diff
@@ -39,19 +39,19 @@
...
-                $content = $_POST['link'];
+                $content = $_POST['link'] ?? '';
             }
-            $name = $_POST['name'];
-            $category = $_POST['category'];
-            $purpose = $_POST['purpose'];
+            $name = $_POST['name'] ?? '';
+            $category = $_POST['category'] ?? '';
+            $purpose = $_POST['purpose'] ?? '';
             $tags = strtolower($_POST['tags']);
```
Comment (SKuipers) [inline modules/Planner/resources_manage_addProcess.php:53]: 👀

File: modules/Planner/resources_manage_addProcess.php [hunk 12]
```diff
@@ -39,19 +39,19 @@
...
-            $gibbonYearGroupIDList = (!empty($_POST['gibbonYearGroupID']))? implode(',', $_POST['gibbonYearGroupID']) : '';
+            $gibbonYearGroupIDList = implode(',', $_POST['gibbonYearGroupID']) ?? '';
             $description = $_POST['description'];
```
Comment (SKuipers) [inline modules/Planner/resources_manage_addProcess.php:55]: 👀

File: modules/Planner/resources_manage_editProcess.php [hunk 13]
```diff
@@ -69,20 +69,20 @@
...
-                        $content = $_POST['link'];
+                        $content = $_POST['link'] ?? '';
                     }
-                    $name = $_POST['name'];
-                    $category = $_POST['category'];
-                    $purpose = $_POST['purpose'];
+                    $name = $_POST['name'] ?? '';
+                    $category = $_POST['category'] ?? '';
+                    $purpose = $_POST['purpose'] ?? '';
                     $tags = strtolower($_POST['tags']);
```
Comment (SKuipers) [inline modules/Planner/resources_manage_editProcess.php:83]: 👀

File: modules/Planner/resources_manage_editProcess.php [hunk 14]
```diff
@@ -69,20 +69,20 @@
...
-                    $gibbonYearGroupIDList = (!empty($_POST['gibbonYearGroupID']))? implode(',', $_POST['gibbonYearGroupID']) : '';
-                    $description = $_POST['description'];
+                    $gibbonYearGroupIDList = implode(',', $_POST['gibbonYearGroupID']) ?? '';
```
Comment (SKuipers) [inline modules/Planner/resources_manage_editProcess.php:84]: This is outside the function call and will not have the desired effect. The null coalesce needs to be on the $_POST variable itself.

File: modules/Planner/resources_manage_addProcess.php [hunk 15]
```diff
@@ -39,19 +39,19 @@
...
             }
-            $name = $_POST['name'];
-            $category = $_POST['category'];
-            $purpose = $_POST['purpose'];
+            $name = $_POST['name'] ?? '';
+            $category = $_POST['category'] ?? '';
+            $purpose = $_POST['purpose'] ?? '';
             $tags = strtolower($_POST['tags']);
-            $gibbonYearGroupIDList = (!empty($_POST['gibbonYearGroupID']))? implode(',', $_POST['gibbonYearGroupID']) : '';
+            $gibbonYearGroupIDList = implode(',', $_POST['gibbonYearGroupID']) ?? '';
```
Comment (SKuipers) [inline modules/Planner/resources_manage_addProcess.php:54]: This is outside the function call and will not have the desired effect. The null coalesce needs to be on the $_POST variable itself.

File: modules/Planner/units_addProcess.php [hunk 16]
```diff
@@ -125,7 +123,7 @@
                     }
 
                     //ADD BLOCKS
-                    $blockCount = ($_POST['blockCount'] - 1);
+                    $blockCount = ($_POST['blockCount'] - 1) ?? '';
```
Comment (SKuipers) [inline modules/Planner/units_addProcess.php:126]: This is outside the brackets and will not have the desired effect. The null coalesce needs to be on the $_POST variable itself.

File: modules/Planner/units_duplicateProcess.php [hunk 17]
```diff
@@ -124,7 +124,7 @@
 
                 //Copy Lessons & resources
                 if ($copyLessons == 'Y') {
-                    $gibbonCourseClassIDSource = $_POST['gibbonCourseClassIDSource'];
+                    $gibbonCourseClassIDSource = $_POST['gibbonCourseClassIDSource'] ?? '';
                     $gibbonCourseClassIDTarget = null;
                     if (isset($_POST['gibbonCourseClassIDTarget'])) {
```
Comment (SKuipers) [inline modules/Planner/units_duplicateProcess.php:129]: 👀

File: modules/Planner/units_editProcess.php [hunk 18]
```diff
@@ -152,7 +150,7 @@
                         //Update blocks
                         $order = '';
                         if (isset($_POST['order'])) {
```
Comment (SKuipers) [inline modules/Planner/units_editProcess.php:152]: The isset it not required with the addition of a null coalesce.

**Review by SKuipers — CHANGES_REQUESTED** [review]
Hi Harry, I've added a number of comments to this one. There's some spots where null coalesce is missing, and a few spots to check the syntax on it, and be sure to apply it to $_POST itself rather than outside a function call. Give a shout if you have any questions.

File: modules/Planner/units_addProcess.php (hunk 16 above)
Comment (yookoala) [reply modules/Planner/units_addProcess.php:126]: For the calculation to make sense, I think it should be `0` instead of `''`.

**Comment by hmerrettICHK** [comment]
I've made the tweaks you suggested. Could you please take a look and re-review?

File: modules/Planner/resources_add_ajaxProcess.php [hunk 19]
```diff
@@ -34,19 +34,19 @@
...
-        $category = $_POST[$id.'category'];
-        $purpose = $_POST[$id.'purpose'];
-        $tags = strtolower($_POST[$id.'tags']);
-        $gibbonYearGroupIDList = (!empty($_POST[$id.'gibbonYearGroupID']))? implode(',', $_POST[$id.'gibbonYearGroupID']) : '';
-        $description = $_POST[$id.'description'];
+        $name = $_POST[$id.'name'] ?? '';
+        $category = $_POST[$id.'category'] ?? '';
+        $purpose = $_POST[$id.'purpose'] ?? '';
+        $tags = strtolower($_POST[$id.'tags'] ?? '');
+        $gibbonYearGroupIDList = implode(',', $_POST[$id.'gibbonYearGroupID']) ?? '';
```
Comment (SKuipers) [inline modules/Planner/resources_add_ajaxProcess.php:48]: The null coalesce is still outside the function in this instance. It needs to be applied directly to the $_POST variable. Can you take a closer look and update it? Thanks

File: modules/Planner/units_addProcess.php [hunk 20]
```diff
@@ -125,7 +123,7 @@
                     }
 
                     //ADD BLOCKS
-                    $blockCount = ($_POST['blockCount'] - 1);
+                    $blockCount = ($_POST['blockCount'] - 1 ?? 0);
```
Comment (SKuipers) [inline modules/Planner/units_addProcess.php:126]: Not quite there yet. The null coalesce always needs applied directly to the $_POST variable.
```suggestion
                    $blockCount = ($_POST['blockCount'] ?? 0) - 1;
```

**Review by SKuipers — CHANGES_REQUESTED** [review]
Thanks Harry, you're almost there. I've added two more comments to spots in the code to check out.

**Comment by SKuipers** [comment]
Hi Harry, I've pushed a commit to help illustrate a fix for null coalesce in resources_add_ajaxProcess.php , which you can see here: https://github.com/GibbonEdu/core/pull/1262/commits/ba8f9953cf74557c5d16b9ebf8f9e6743bef9015

The difference is that the null coalesce operator needs to be applied directly after the [], and in this case you had placed it inside the []s. Hope this helps clarify the syntax for future changes.

**Review by SKuipers — APPROVED** [review]

**Pull request merged**